datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/DatalakeDrClient.java

```diff
@@ -111,16 +139,65 @@ public DatalakeDrStatusResponse getRestoreStatusByRestoreId(String datalakeName,
 
         try (ManagedChannelWrapper channelWrapper = makeWrapper()) {
             RestoreDatalakeStatusRequest.Builder builder = RestoreDatalakeStatusRequest.newBuilder()
-                .setDatalakeName(datalakeName)
-                .setRestoreId(restoreId);
+                    .setDatalakeName(datalakeName)
+                    .setRestoreId(restoreId);
 
             return statusConverter.convert(
-                newStub(channelWrapper.getChannel(), UUID.randomUUID().toString(), actorCrn)
-                    .restoreDatalakeStatus(builder.build())
+                    newStub(channelWrapper.getChannel(), UUID.randomUUID().toString(), actorCrn)
+                            .restoreDatalakeStatus(builder.build())
             );
         }
     }
 
+    public DatalakeBackupInfo getLastSuccessBackup(String datalakeName, String actorCrn) {
+        DatalakeBackupInfo datalakeBackupInfo = null;
+        if (!datalakeDrConfig.isConfigured()) {
+            return null;
+        }
+
+        checkNotNull(datalakeName);
+        checkNotNull(actorCrn);
+
+        try (ManagedChannelWrapper channelWrapper = makeWrapper()) {
+            ListDatalakeBackupRequest.Builder builder = ListDatalakeBackupRequest.newBuilder()
+                    .setDatalakeName(datalakeName);
+            ListDatalakeBackupResponse response = newStub(channelWrapper.getChannel(), UUID.randomUUID().toString(), actorCrn)
+                    .listDatalakeBackups(builder.build());
+
+            if (response != null && response.getDatalakeInfoList() != null) {
+                datalakeBackupInfo = response.getDatalakeInfoList().stream()
+                        .filter(backup -> "SUCCESSFUL".equals(backup.getOverallState()))
+                        .findFirst().orElse(null);
```

Comment:
Using the first entry does not guarantee that it is the latest. We need to use a sorted list.

Reply:
Yes. It does. Because thunderhead already sort it. 

backups = datalakeDrRepository.getAllBackups(getAccountId(), request.getDatalakeName());


```
  public List<DatalakeDrStatusEntity> getAllBackups(String accountId, String datalakeName) {
    return datalakeDrRepository.getDatalakeDrStatusEntitiesSorted(accountId, datalakeName, DatalakeDrStatusEntity.DrOperation.BACKUP);
  }
```


"select entity from DatalakeDrStatusEntity entity where entity.accountId = :accountId" +
      " and entity.datalakeName = :datalakeName and entity.operation = :operation **order by entity.startTime desc"**

Eventually, it goes to this SQL in thunderhead that the returned list is sorted by starttime desc. So the first one is the latest. 

I think it is not necessary to sort again in cloudbreak. But I can sort again. Just feel not necessary.

Reply:
You are right.